plugins/aea-ledger-fetchai/aea_ledger_fetchai/_cosmos.py

```diff
@@ -784,17 +748,26 @@ def _get_storage_transaction(
         :param builder: the builder.
         :return: the unsigned CosmWasm contract deploy message
         """
-        store_msg = {
-            "type": MESSAGE_FORMAT_BY_VERSION["store"][self.cosmwasm_version],
-            "value": {
-                "sender": deployer_address,
-                "wasm_byte_code": contract_interface[_BYTECODE],
-                "source": source,
-                "builder": builder,
-            },
-        }
+        store_msg = MsgStoreCode(
+            sender=str(deployer_address),
+            wasm_byte_code=base64.b64decode(contract_interface[_BYTECODE]),
+            source=source,
+            builder=builder,
+        )
+        store_msg_packed = ProtoAny()
+        store_msg_packed.Pack(store_msg, type_url_prefix="/")
+
+        tx_fee_coins = [Coin(denom=denom, amount=str(tx_fee))]
         tx = self._get_transaction(
-            account_number, chain_id, tx_fee, denom, gas, memo, sequence, msg=store_msg,
+            account_numbers=[account_number],
+            from_addresses=[str(deployer_address)],
+            pub_keys=[b""],
+            chain_id=chain_id,
+            tx_fee=tx_fee_coins,
+            gas=gas,
+            memo=memo,
+            sequences=[sequence],
+            msgs=[store_msg_packed],
```

Comment:
clarify the diffs...

Reply:
account numberS
from_addresses
public_keys
denom(missing)
sequenceS
msgS

Reply:
Looks like I just forgot to copy things over.

Reply:
Actually I moved this line down: amount_coins = [Coin(denom=denom, amount=str(amount))]
because cosmos tests are using Mocks which fails to be converted to amount - it expects it to be integer as string. And these cosmos tests make get account number and sequence to fail so it won't get past this point.